src/models/pieces/pawn.mjs

```diff
@@ -83,19 +93,47 @@ class Pawn extends Piece {
       available.push(Move.fromSquare(secondSquare, this));
     }
 
+    return available;
+  }
 
+  setPromote(move) {
+    if (move.row === 0 || move.row === 7) {
+      this.isPromoted = true;
+    }
+  }
 
-    return available;
-;
+  promote(move, type) {
+    move.player.removeLivePiece(this);
+    move.player.addPromotedPawn(this);
+
+    const pieceConstuctors = {
```

Comment:
this mapping can live outside of this function, since it will always be the same